mx.h

```diff
@@ -184,6 +210,10 @@ struct MxOps
    * @param msgno Index of message to open
    * @retval  0 Success
    * @retval -1 Error
+   *
+   * **Contract**
+   * - @a m   is not NULL
+   * - @a msg is not NULL
```

Comment:
What about `msgno`?
Perhaps: `0 <= msgno < m->msg_count`

Reply:
I generally only added contract for existing invariants, I didn't add more checks (minus the two cases where that was the only sensible input)

I can do that, but then we should probably fail in mx.c if the invariant isn't met right?